certbot/tests/main_test.py

```diff
@@ -1164,6 +1165,69 @@ def test_update_registration_with_email(self, mock_utility, mock_email):
                             email in mock_utility().add_message.call_args[0][0])
 
 
+class UnregisterTest(unittest.TestCase):
+    def setUp(self):
+        self.patchers = {
+            '_determine_account': mock.patch('certbot.main._determine_account'),
+            'account': mock.patch('certbot.main.account'),
+            'client': mock.patch('certbot.main.client'),
+            'getUtility': mock.patch('certbot.main.zope.component.getUtility')}
```

Comment:
This is new and you had no way of knowing, but `certbot.test.util.patch_get_utility` should be used here. It does additional error checking on methods like `yesno` to try and catch bugs.

`test_abort_unregister` will have to be modified slightly. It uncovered a bug which you can see in #4071. For now, don't set the `getUtility` return value. You can set properties of it, but don't change the object itself.

Stuff like this is OK:
```
display = self.mocks['getUtility'].return_value
display.yesno.return_value = False
self.mocks['getUtility'].return_value.foo.return_value = False
```
At least for now, anything like this isn't:
```
self.mocks['getUtility'].return_value = ...
```